### PR TITLE
AUSM+ -up2 solver implementation

### PR DESCRIPTION
**Pull request by aeroamit**
Based on paper - 

## Proposed Changes

- AUSM+ -up2 scheme has been implemented

- Likely to perform better for high speed flows (author claims - Robustness against shock anomalies as good as Van leer scheme (less carbuncle prone) while it resolves contact discontinuity/boundary layer sharply/accurately.  Smooth representation of heating etc.)

- Reference- "Towards a shock-stable and accurate hypersonic heating computations: A new pressure flux for AUSM scheme, authors- Kitamura, Shima,  JCP 245 2013"

## Related Work
*Resolve any issues (bug fix or feature request), note any related PRs, or mention interactions with the work of others, if any.*

## PR Checklist
*Put an X by all that apply. You can fill this out after submitting the PR. If you have any questions, don't hesitate to ask! We want to help. These are a guide for you to know what the reviewers will be looking for in your contribution.*

- [ X ] I am submitting my contribution to the develop branch.
- [ X ] My contribution generates no new compiler warnings (try with the '-Wall -Wextra -Wno-unused-parameter -Wno-empty-body' compiler flags).
- [ X ] My contribution is commented and consistent with SU2 style.
- [  ] I have added a test case that demonstrates my contribution, if necessary.

### CODE REVIEW

**Comment by pcarruscag** [comment]
Hi @aeroamit ,
Thank you for the work you have been doing on AUSM schemes.
This and your previous PR seem to share a lot of the code, do you think it is possible to generalize this? i.e. boil down AUSM schemes to just a change in the constants of the method, see for example what has been done in #626.
Cheers,
Pedro

**Comment by aeroamit** [comment]
Hi @pcarruscag,
You are right, it shares a lot of code with previous PR. Here constants does not change. Differences are as follows - 

1- It has different pressure flux definition (new expression)
2 - Here sum of squares of velocity components were needed for left and right state
3- removed few variables and added few.
4- Also left and right state split Mach numbers are Mach number polynomials only (for programming purpose) without pressure terms.

Previously SLAU and SLAU2 scheme have already been implemented in the code separately. I mean some of these sucessive schemes share a lot of common formulation but differ with some expressions, constants etc. 
So it may be fine to keep them separate. 

I will see your advice and further look into similar variation implementation in the code

Thanks 
Amit

**Comment by aeroamit** [comment]
Hi @EduardoMolina  ,
I have checked the code/paper so -

**SLAU2 = SLAU+modified pressure flux**   &

**AUSM+-up2 = AUSM+ -up + SLAU2 modified pressure flux**

So AUSM+-up2 implementation differs from SLAU2 (page no. 72-73 of reference  - Towards shock-stable and accurate hypersonic heating computations: A new pressure flux for AUSM-family schemes, KeiichiKitamura, EijiShima,  245 (2013) 62–83).

Also I am just copying the lines from the paper  (page no. - 81) - "....Thus, for **practical usage**, it is advised that the **users adopt AUSM+-up2** or **LDFSS2001-2** for **hypersonic flows** or low speed flows when reference velocity is available, whereas **SLAU2** is **recommended for low speed flows** **without reference velocity**; either of these three can be used for moderate speeds, as they are exactly the same as their original fluxes ....."

Hence AUSM+-up2 can be a good candidate for usage.

I think this resolves the issue @EduardoMolina .

Cheers.
Amit

**Comment by economon** [comment]
Thanks for the technical exchange, @aeroamit @EduardoMolina, this is nice to see.

@aeroamit : do you have any specific cases that use the different schemes you are submitting? Could be good to see the differences and also use them for regression testing (our matrix could prob be better at covering more convective schemes).

**Comment by aeroamit** [comment]
Hi @economon ,

I tried existing Test cases from Repository (NACA0012 etc.). At lower Mach numbers, there is improvement in accuracy over AUSM, HLLC or Roe scheme (especially stagnation region values and there is some more story to it). 
I will try to workout some specific test case for higher Mach number to show the differences.
Also some of these latest schemes are likely to be better in comparison to existing ones/predecessors in most of the situations (incrementally or reasonably).

Regards 
Amit

**Comment by aeroamit** [comment]
Hi @economon @EduardoMolina 

I have tried two test cases - NACA 0012 airfoil and flow past blunt body. Please see the attached pdf file containing the details.

[AUSM_UP_UP2.pdf](https://github.com/su2code/SU2/files/2746891/AUSM_UP_UP2.pdf)

Regards
Amit

**Comment by EduardoMolina** [comment]
Hi @aeroamit 

Thank you for your time and for share this great material. I really appreciated.

I will review your pull request in a couple of days.

Best,

Eduardo

**Comment by EduardoMolina** [comment]
Hi @aeroamit ,

LGTM! Thanks.

Just a mark: I think it is interesting to add the flow past a blunt body on the test repo. What do think? If not, it is ready to merge.

Eduardo

**Comment by aeroamit** [comment]
Hi @EduardoMolina 

I will add this in the test repo. It is a standard case used in majority of the papers (relevant).
What are the changes/additions I need to carry out , can you guide me with the procedure of adding the test case in repo.

Regards
Amit

**Comment by EduardoMolina** [comment]
Hi @aeroamit 

Sure. Please see this link: https://su2code.github.io/docs/Gitting-Started/

Basically, you need to modify the python regression files in order to add your new test case. Also, you need to upload the grid in https://github.com/su2code/TestCases, you can create a branch for that. Lastly, update .travis.yml but don't forget to revert it before merge.

Best,
Eduardo

**Comment by aeroamit** [comment]
Hi @EduardoMolina

I have 1- created a folder and added a .cfg file (TestCases/euler/bluntbody/blunt.cfg) 2- added details in serial as well as parallel_regression.py 3- Added the mesh TestCases/euler/bluntbody/blunt_91.su2 in online repository (merged with develop branch).

Now Travis CI is throwing following error- 
Traceback (most recent call last):
  File "serial_regression.py", line 1403, in <module>
    main()
  File "serial_regression.py", line 129, in main
    test_list.append(bluntbody)
NameError: global name 'bluntbody' is not defined
The command "python $TEST_SCRIPT" exited with 1.  (In parallel_regression as well)
(I missed out some small changes??)

Regards
Amit

**Comment by aeroamit** [comment]
Travis CI check is hanging. I am not able to find a way out.
Help please...... 

Thanks 
Amit

**Comment by aeroamit** [comment]
Hi @EduardoMolina

Submitted the PR for Testcase (mesh file)
Thanks Amit

**Comment by EduardoMolina** [comment]
Can you update this branch to develop? I think now Travis will run fine.

Eduardo

**Comment by EduardoMolina** [comment]
Hi SU2 developers,

Can someone shed a light on why Travis is "Waiting for status to be reported"?
I would like to merge this PR ASAP.

Best,
Eduardo

**Comment by aeroamit** [comment]
Hi @EduardoMolina

I am not able to figure out a solution for this online. 
Should we wait for some community member to resolve this or shall I submit a copy of the same as a fresh pull request and close this one ? 

Best,
Amit

**Comment by economon** [comment]
Hi @EduardoMolina @aeroamit : I think it is running now. I don't think the tests could execute because the travis file was pointing to a branch in an external fork. I have reverted it to develop.

**Comment by aeroamit** [comment]
Thanks @economon

**Comment by EduardoMolina** [comment]
Thanks @economon ....

@aeroamit : Can you please check the values of the blunt body test case? I think the test case you added failed.

Best.
Eduardo

**Comment by aeroamit** [comment]
Hi @EduardoMolina

Actually I pasted the values for iteration no 21 instead of 20 that's why it failed. 
I am correcting it and also my output files are attached for reference.

[serialop.pdf](https://github.com/su2code/SU2/files/2826414/serialop.pdf)
[parallelop.pdf](https://github.com/su2code/SU2/files/2826416/parallelop.pdf)

Regards
Amit

File: .travis.yml [hunk 1]
```diff
@@ -12,7 +12,7 @@ compiler:
 notifications:
     email:
         recipients:
-            - su2code-dev@lists.stanford.edu
```
Comment (EduardoMolina) [inline .travis.yml]: Please revert this back to su2code list.

**Comment by EduardoMolina** [comment]
Hi @aeroamit, 

Please revert back *.travis.yml to su2code mail list and update to develop....
Once all checks passed again, I will merge it ASAP.

Thanks,
Eduardo

**Comment by EduardoMolina** [comment]
Thanks @aeroamit for the contribution....
Merging now.
Eduardo

**Pull request merged**